src/python/pants/base/run_info.py

```diff
@@ -66,7 +67,10 @@ def add_infos(self, *keyvals, **kwargs):
       if k in self._info:
         raise ValueError('info key "{}" already exists with value {}. '
                          'Cannot add it again with value {}.'.format(k, self._info[k], v))
-      self._info[k] = v
+      if k == 'target_data':
```

Comment:
Might be nicer to keep RunInfo simple, and move this to the caller, which already knows when to do this eval, instead of having to know about this magic key string here?